code/headers/robot_arm_interface_c.hpp

```diff
@@ -0,0 +1,22 @@
+//
+// Created by Slimmerik on 13/04/2019.
+//
+
+#pragma once
+
+#include <vector3_c.hpp>
+#include <hwlib.hpp>
+
+namespace r2d2::robot_arm{
+     class robot_arm_interface{
+
+     public:
+         // pure virtual function providing interface framework.
+         virtual void move_head_to_coordinate(const vector3_c< uint16_t > &v) = 0;
```

Comment:
use a more descriptive variable name and use type vector3i_c instead of vector3_c<uint16_t>

Reply:
Also no spaces between angle brackets, use `<uint16_t>` instead of `< uint16_t >`